Firestore/Source/API/FIRFirestore+Internal.h

```diff
@@ -34,7 +34,8 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithProjectID:(NSString *)projectID
                          database:(NSString *)database
                    persistenceKey:(NSString *)persistenceKey
-              credentialsProvider:(id<FSTCredentialsProvider>)credentialsProvider
+              credentialsProvider:(firebase::firestore::auth::CredentialsProvider *)
```

Comment:
This is an internal header, so if you like, you can do: `using firebase::firestore::auth::CrednetialsProvider;`

Optional.

Reply:
I'm not a fan of using in headers because it bleeds out beyond the header into all implementations, not just the .cc or .mm paired that implements this header. This wouldn't be much of a problem except there's no way to un-using a symbol and removing a using declaration later has the potential to change ADL.